scripts/oneoff/fix-agreement-extensions.py

```diff
@@ -0,0 +1,127 @@
+"""
+
+Usage:
+    scripts/oneoff/fix-agreement-extensions.py <stage> <api_token> <download_directory> [--dry-run]
+"""
+import sys
+sys.path.insert(0, '.')
+
+import os
+import re
+
+import magic
+from docopt import docopt
+
+from dmutils.apiclient import DataAPIClient
+from dmutils.s3 import S3
+from dmscripts.env import get_api_endpoint_from_stage
+
+
+def get_all_pdfs(download_directory):
+    for root, subfolder, files in os.walk(download_directory):
+        for filename in files:
+            if filename.endswith('.pdf') and not filename.startswith('2015-11-'):
+                yield os.path.join(root, filename)
+
+
+def get_filetype(path):
+    return magic.from_file(path, mime=True)
+
+
+def is_empty(path):
+    stat = os.stat(path)
+    return stat.st_size == 0
+
+
+def get_supplier_id_from_path(path):
+    match = re.search(r'/(\d+)/', path)
+    if not match:
+        raise ValueError("Could not find supplier ID in path {}".format(path))
+    return match.group(1)
+
+
+def handle_path(client, bucket, dry_run, path):
+    if is_empty(path):
+        show_contact_details(client, dry_run, path)
+    else:
+        filetype = get_filetype(path)
+        if filetype != b"application/pdf":
+            update_file_extension(client, bucket, dry_run, path, filetype)
+
+
+def show_contact_details(client, dry_run, path):
+    supplier_id = get_supplier_id_from_path(path)
+    if dry_run:
+        print("Empty file for {} - {}".format(supplier_id, os.path.basename(path)))
+    else:
+        supplier = client.get_supplier(supplier_id)['suppliers']
+        print(
+            "Empty file for {} {}".format(
+                supplier_id,
+                supplier['contactInformation'][0]['email']))
```

Comment:
I don't think this is the best email address to use - this is the email address shown to buyers who might want to buy something.  I reckon the Digital Marketplace user account email addresses for the supplier would be a better bet.

Some suppliers also specified a different email in their declaration as the "Primary contact for the tender" from *any* of their marketplace account emails (I know!), so it might be worth pulling the primary email from the declaration (it's `SQ1-2b`) as well as the users.